res/res_odbc.c

```diff
@@ -971,17 +972,17 @@ struct odbc_obj *_ast_odbc_request_obj2(const char *name, struct ast_flags flags
 			/* If the connection is dead try to grab another functional one from the
 			 * pool instead of trying to resurrect this one.
 			 */
-			ao2_ref(obj, -1);
-			obj = NULL;
-
 			ast_mutex_lock(&class->lock);
 
 			class->connection_cnt--;
+			/* ast_cond_signal(&class->cond); - this thread will re-acquire, and that fails will signal */
```

Comment:
Remove the ast_cond_signal in even the comment and just explain in a comment, I also think you missed an "if": "and if that fails"

Reply:
Should be better now?